mlflow/utils/checkpoint_utils.py

```diff
@@ -114,7 +114,6 @@ def check_and_save_checkpoint_if_needed(self, current_epoch, global_step, metric
             if not self._is_new_checkpoint_better(new_monitor_value):
                 # Current checkpoint is worse than last saved checkpoint,
                 # so skip checkpointing.
-                self.last_monitor_value = new_monitor_value
```

Comment:
Thanks for the PR! https://github.com/mlflow/mlflow/pull/11197 is the original PR that implemented automatic checkpoint selection. Wondering if we could improve the tests to ensure the logic to select the best model works correctly 🤔